### ISSUE TITLE
feat(Stepper): Disable header navigation

### ISSUE THREAD

**Issue by dev054** [angular/components#20427]
#### Feature Description

Currently there's no fancy way to disable header navigation.

#### Use Case

In a stepper you may want to control it through buttons "Back" and "Next" only.

**Comment by crisbeto** [comment]
I can sort of see the use case, but it's unclear how we'd handle the accessibility in this case. That being said, you should be able to achieve the same by adding this to your CSS:

```css
.mat-step-header {
  pointer-events: none;
}
```

**Comment by LuisOfCourse** [comment]
Hi everyone,

just to let u know that this feature is really important for us.
The fact that this Issue isn't crowded of people asking for a change is for a easy fix offered on [a StackOverflow question](https://stackoverflow.com/questions/46933953/angular-material-stepper-disable-header-navigation).

Please, offer some kind of way to set this component to something like: 
`<mat-stepper [navigationHeader]="false">`

**Comment by BotInterpreters** [comment]
yes, we need this option for cases for stepper with progress bar if they navigate to by using the header navigation the incrementing of progress bar of using next and back button will not be used

**Comment by pouyababaie** [comment]
i faced the same issue here binding reactive form. why there is an error state and a ``` stepControl``` but there is no logic for disabling the navigation if there an error in the forms? 
ias @LuisOfCourse  , it is not a very major change.

**Comment by LuisOfCourse** [comment]
This is not a major change but the original request dates back to 2020. A lot of time ago.
For other reasons I created my custom stepper and other things that are a replacement of material.

**Comment by pouyababaie** [comment]
@LuisOfCourse 
yeah , i may do it myself too.
